### DISCUSSION TITLE
Hierarchical time series forecasting and machine learning algorithms

### DISCUSSION THREAD

**Discussion by antoniosudoso** [Nixtla/hierarchicalforecast#126] [Ideas]
There is an increasing amount of literature on machine-learning algorithms for forecasting HTS and generating coherent forecasts. 
For future releases, it would be nice to provide a Python implementation for them. In chronological order we have:

1) [A machine learning approach for forecasting hierarchical time series (2021)](https://www.sciencedirect.com/science/article/abs/pii/S0957417421005431)
2) [Hierarchical forecast reconciliation with machine learning (2021)](https://www.sciencedirect.com/science/article/abs/pii/S1568494621006773)
3) [Machine learning applications in hierarchical time series forecasting: Investigating the impact of promotions (2022)](https://www.sciencedirect.com/science/article/abs/pii/S0169207022001029)

**Comment by kdgutier** [comment]
Thanks for the pointer, 
It would be cool to have them implemented in python too.
Do you have pointers to the original code?
